Projects Refinements

sandeepsajan0 opened the pull request:
Fixes #3333 
Fixes #3352 
FIxes retesting issues of #3331

sandeepsajan0 committed:
Fix Supporting document view permissions for paf approvers

sandeepsajan0 committed:
Show questions of stream field form even if there is no answer

sandeepsajan0 committed:
Fix thread retest issues(partial)

sandeepsajan0 committed:
Add caret up/down icons to completed stage like Project documents and contracting documents

sandeepsajan0 commented:
@frjo  I think we can put this on test, it contains retest UI fixes, Stream form's field questions(include without answer), and other small fixes.

frjo commented:
@sandeepsajan0 Being deployed to test now.

theskumar commented on hypha/apply/projects/templates/application_projects/includes/supporting_documents.html:
```diff
@@ -4,7 +4,12 @@
 
 <div class="docs-block wrapper--outer-space-large" {% if collapsible_header %} x-data="{ collapsed: false }" {% endif %}>
     <div class="docs-block__header" {% if collapsible_header %} x-on:click="collapsed = ! collapsed" {% endif %}>
```
As this is non-standard actionable html element please add aria-label and `role=button` to it.

theskumar replied:
see this for more details: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/

frjo replied:
When possible I try to use `<details><summary></summary></details>` HTML-tags for collapsible content. Then the browser handels it all. Might not work in this instance.

theskumar replied:
I believe it should be possible here.

sandeepsajan0 replied:
As it is a conditional accordion so `details` and `summary` tag doesn't make much sense here(it acts as an accordion only after a stage). But yes for the accessibility I agree I would add aria-label and aria-expanded and role to it.

theskumar commented on hypha/apply/projects/templates/application_projects/includes/supporting_documents.html:
```diff
@@ -4,7 +4,12 @@
-        <div class="docs-block__heading">{% trans "Project documents" %}</div>
+        <div class="docs-block__heading">{% trans "Project documents" %}
+            {% if collapsible_header %}
+            <svg class="icon icon--caret-up" x-show="collapsed"><use xlink:href="#caret-up"></use></svg>
```
Add `aria-hidden=true` for this `svg`

theskumar commented on hypha/apply/projects/templates/application_projects/project_approval_detail.html:
```diff
@@ -156,7 +157,7 @@ <h5>{% trans "Actions to take" %}</h5>
                     {% endif %}
                     <div class="dropdown">
-                      <button class="button button--white dropbtn button--bottom-space button--full-width" type="button" data-dropdown-target="#dropdown-content-download">
+                      <button class="button button--white dropbtn button--bottom-space button--full-width" style="padding:0.5em 53px" type="button" data-dropdown-target="#dropdown-content-download">
```
Let's not use custom spacing unless it's absolutely needed. Use `tailwind-css` classes for the spacing. 

Here you can use `py-1 py-3`, also the current default padding on the buttons is too much, i've modified it here in my PR, you can also change the default padding: 

https://github.com/HyphaApp/hypha/pull/3149/files#diff-6ae81798f0bc08223ae8def2462a5eed782ace1b07ef3d4da63a7e21ee8be5e8R81

![image](https://user-images.githubusercontent.com/236356/232687064-da98dee9-e721-440d-822e-6e5eec7b47b4.png)


hypha/static_src/src/sass/apply/abstracts/_mixins.scss

sandeepsajan0 replied:
I was thinking to convert them into a css class after testing but I think it might be better if we can use tailwind classes directly. I'll try that out.

sandeepsajan0 committed:
fix data - server issue for stream field render without answers

theskumar commented on hypha/apply/projects/templatetags/project_tags.py:
```diff
@@ -147,3 +151,12 @@ def user_can_remove_supporting_documents(project, user):
+
+
+@register.simple_tag
+def user_can_take_actions(project, user):
```
Can you add a docstring to this function to explain where it used. the `user_can_take_actionss` seems too generic to me here. in the `project_tags.py` context.

sandeepsajan0 commented:
@frjo  For #3352, I have returned '-' (dash) previously but there was some issue with this, and the submission detail pages were throwing 500 errors. So, I changed it to '' (empty string) in the latest commit and now it is working fine and showing "No Response" text that is configured in the code(at multiple places), so we can change it. 
We have to decide what we want to use in place of "No Response" as text.

sandeepsajan0 committed:
Add suggested accecibility to project and contract accordions

sandeepsajan0 committed:
Use tailwind classes instead of inline styles

sandeepsajan0 committed:
Add doc strings to template tags

sandeepsajan0 committed:
Invoice update status button in invoice row on project detail page (#3357)

Add 'Update status' button for invoices on the project detail page for
staff and finance(who can update status).

sandeepsajan0 committed:
Remove paf download button inline css

sandeepsajan0 committed:
Add delete button to invoice edit page

frjo commented:
Deployed to test now.

sandeepsajan0 committed:
fix latest test issues, and change delete button colour to red on file upload

frjo commented:
@sandeepsajan0 Should I merge this PR and we can work on remaining issue in a new one?

sandeepsajan0 commented:
@frjo Yes, we can merge it in.
